### PR TITLE
Remove not used database dialect property

### PR DESCRIPTION
**Pull request by dacrome**

### CODE REVIEW

**Comment by tkrille** [comment]
Can you do this in the ITs too, please?

**Comment by tkrille** [comment]
The property must be removed in the docs too: https://github.com/osiam/resource-server/blob/master/docs/configuration.md. Might be also true for the docs in osiam/osiam.

**Comment by tkrille** [comment]
And I think, that a changelog entry is mandatory for this kind of change.

**Comment by tkrille** [comment]
I've grep'ed my local sources:

```
osiam-osiam/docs/detailed-reference-installation.md:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
test-suites/performance-tests/src/main/resources/resource-server-configuration/jpa-configuration.xml:                 <property name="databasePlatform" value="${org.osiam.resource-server.db.dialect}"/>
test-suites/performance-tests/src/main/resources/resource-server-configuration/resource-server_pg.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
test-suites/performance-tests/src/main/resources/resource-server-configuration/resource-server_h2.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.H2Dialect
resource-server/docs/configuration.md:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
osiam-connector4java-integration-tests/src/test/resources/resource-server-configuration/resource-server_mysql.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
osiam-connector4java-integration-tests/src/test/resources/resource-server-configuration/resource-server_pg.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
```

**Comment by dacrome** [comment]
> Can you do this in the ITs too, please?

Yes, here you go: https://github.com/osiam/connector4java-integration-tests/pull/271

**Comment by dacrome** [comment]
> I've grep'ed my local sources:
> osiam-osiam/docs/detailed-reference-installation.md:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
> test-suites/performance-tests/src/main/resources/resource-server-configuration/jpa-configuration.xml:                 <property name="databasePlatform" value="${org.osiam.resource-server.db.dialect}"/>
> test-suites/performance-tests/src/main/resources/resource-server-configuration/resource-server_pg.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
> test-suites/performance-tests/src/main/resources/resource-server-configuration/resource-server_h2.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.H2Dialect
> resource-server/docs/configuration.md:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect
> osiam-connector4java-integration-tests/src/test/resources/resource-server-configuration/resource-server_mysql.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
> osiam-connector4java-integration-tests/src/test/resources/resource-server-configuration/resource-server_pg.properties:org.osiam.resource-server.db.dialect=org.hibernate.dialect.PostgresPlusDialect

This is true for all except the `test-suites`. `test-suites` don't use a current version of the `resource-server`.

**Comment by dacrome** [comment]
> The property must be removed in the docs too: https://github.com/osiam/resource-server/blob/master/docs/configuration.md. Might be also true for the docs in osiam/osiam.

Done.

**Comment by dacrome** [comment]
> And I think, that a changelog entry is mandatory for this kind of change.

Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - Update OSIAM connector4java
 - Remove support for old, method-based OAuth scopes
+- Remove not used database dialect property
```
Comment (tkrille) [inline CHANGELOG.md]: Suggestion: "Remove configuration property `org.osiam.resource-server.db.dialect`"

**Comment by dacrome** [comment]
Done

**Pull request merged**